plugin/Gui/AdminPageSettingsBase.php

```diff
@@ -246,7 +246,25 @@ public function ajax_action()
 			$this->updateValues($row, $pResultObject, $recordId);
 		}
 
-		$pResultObject->messageKey = $this->getResponseMessagekey($pResultObject->result);
+		$pResultObject->messageKey = $this->getResponseMessagekey( $pResultObject->result );
+		if ( array_key_exists( 'oo_plugin_listviews', $row ) ) {
+			$pResultObject->shortcodeContent = '<span class="viewusage">' . esc_html__( 'Shortcode: ',
+					'onoffice-for-wp-websites' ) . '<input style="margin-right: 5px;" type="text" readonly="" value="[oo_estate view=&quot;' . $row['oo_plugin_listviews']['name'] . '&quot;]"></span>';
+			if ( wp_is_using_https() ) {
```

Comment:
There is an even better way to do the progressive enhancement: You can check if you have access to the necessary functions in JS.

So you would always load a JS-script that would check if it can actually use `navigator.clipboard.writeText()`, and if it can, it will add the button. That means that if anything goes wrong (JS not loaded, or `navigator.clipboard` not available), the button will not be shown. It will only be present if it actually works.

Could you try that?